specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,429 @@
+  COREWEBVIEW2_SAVE_AS_REQUESTED_INVALID_PATH,
+  /// Could not perform Save As because the destination file path already exists and 
+  /// replacing files was not allowed by the `AllowReplace` property.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_FILE_ALREADY_EXISTS,
+  /// Could not perform Save As when the `Kind` property selection not
+  /// supported because of the content MIME type or system limits
+  ///
+  /// MIME type limits please see the emun `COREWEBVIEW2_SAVE_AS_REQUESTED_KIND`
+  ///
+  /// System limits might happen when select `HTML_ONLY` for an error page,
+  /// select `COMPLETE` and WebView running in an App Container, etc.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_NOT_SUPPORTED,
+  /// Did not perform Save As because the client side decided to cancel.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_CANCELLED,
+  /// Save as requested completeed, the downloading would start
+  COREWEBVIEW2_SAVE_AS_REQUESTED_COMPLETED,
+} COREWEBVIEW2_SAVE_AS_REQUESTED_RESULTS;
+
+[uuid(15e1c6a3-c72a-4df3-91d7-d097fbec3bfd), object, pointer_default(unique)]
+interface ICoreWebView2_20 : IUnknown {
+  /// Programmatically trigger a save as action for current content. `SaveAsRequested` 
+  /// event will be raised.
+  ///
+  /// Opens a system modal dialog by default. If it was already opened, this method 
+  /// would not open another one. If the `SuppressDefaultDialog` is TRUE, won't open 
+  /// the system dialog.
+  ///
+  /// The method can return a detailed info to indicate the call's result. 
+  /// Please see COREWEBVIEW2_SAVE_AS_REQUESTED_RESULTS
+  ///
+  /// \snippet ScenarioSaveAs.cpp ProgrammaticSaveAs
+  HRESULT SaveContentAs([in] ICoreWebView2RequestSaveAsCompletedHandler* handler);
+
+  /// Add an event handler for the `SaveAsRequested` event. This event is raised
+  /// when save as is triggered, programmatically or manually.
+  ///
+  /// \snippet ScenarioSaveAs.cpp ToggleEventHandler
+  HRESULT add_SaveAsRequested(
+   [in] ICoreWebView2SaveAsRequestedEventHandler* eventHanlder,
+   [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_SaveAsRequested`.
+  ///
+  /// \snippet ScenarioSaveAs.cpp ToggleEventHandler
+  HRESULT remove_SaveAsRequested(
+   [in] EventRegistrationToken token);
+}
+
+/// The event handler for the `SaveAsRequested` event.
+[uuid(55b86cd2-adfd-47f1-9cef-cdfb8c414ed3), object, pointer_default(unique)]
+interface ICoreWebView2SaveAsRequestedEventHandler : IUnknown {
+  HRESULT Invoke(
+   [in] ICoreWebView2* sender,
+   [in] ICoreWebView2SaveAsRequestedEventArgs* args);
+}
+
+/// The event args for `SaveAsRequested` event
+[uuid(80101027-b8c3-49a1-a052-9ea4bd63ba47), object, pointer_default(unique)]
+interface ICoreWebView2SaveAsRequestedEventArgs : IUnknown {
+  /// Get the Mime type of content to be saved
+  [propget] HRESULT ContentMimeType([out, retval] LPWSTR* value);
+
+  /// Indicates if client side cancelled the silent save as, TRUE means cancelled.
```

Comment:
This wording is confusing. The meaning of client side is not clear in this context and to the end developer. In this case the only one who will set Cancel is the end developer writing the event handler. So you can say 'You can set this to TRUE to cancel the Save As. The default Save As dialog will not be shown and no download will start.'